tt_metal/hw/inc/wormhole/noc_nonblocking_api.h

```diff
@@ -66,6 +66,14 @@ inline uint32_t NOC_CFG_READ_REG(uint32_t noc, uint32_t reg_id) {
   return *ptr;
 }
 
+inline __attribute__((always_inline)) bool noc_cmd_buf_ready(uint32_t noc, uint32_t cmd_buf) {
+  return (NOC_CMD_BUF_READ_REG(noc, cmd_buf, NOC_CMD_CTRL) == NOC_CTRL_STATUS_READY);
+}
+
+inline __attribute__((always_inline)) bool noc_cmd_buf_ready_l1(uint32_t noc, uint32_t cmd_buf) {
+  return (NOC_CMD_BUF_READ_REG_L1(noc, cmd_buf, NOC_CMD_CTRL) == NOC_CTRL_STATUS_READY);
+}
```

Comment:
why do we need both?

Reply:
because these are replacing the corresponding fast_read/write_ok and fast_read/write_ok_l1 resp.